### PR TITLE
[Windows] Migrate error logging to FML_LOG

### PR DESCRIPTION
**Pull request by cbracken**
Migrates error logging from logging directly to stderr to using the
FML_LOG macro with a specified log level.

No additional tests since there is no semantic change to the logging
(FML_LOG simply [writes to stderr](https://github.com/flutter/engine/blob/643742e4fb97900e39b27c704e71bfbab8cce58b/fml/logging.cc#L125-L126) on desktop).

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by a-wallen — APPROVED** [review]

**Comment by yaakovschectman** [comment]
Does `FML_LOG` not require a newline? I notice the old `std::cerr` usages get an `std::endl` at the end, but there's nothing corresponding in the changed versions.

File: shell/platform/windows/flutter_windows_engine.cc [hunk 1]
```diff
@@ -454,7 +454,7 @@ bool FlutterWindowsEngine::SendPlatformMessage(
         embedder_api_.PlatformMessageCreateResponseHandle(
             engine_, reply, user_data, &response_handle);
     if (result != kSuccess) {
-      std::cout << "Failed to create response handle\n";
+      FML_LOG(ERROR) << "Failed to create response handle\n";
```
Comment (loic-sharma) [inline shell/platform/windows/flutter_windows_engine.cc:457]: Should the new line be dropped?

```suggestion
      FML_LOG(ERROR) << "Failed to create response handle";
```

File: shell/platform/windows/keyboard_key_handler.cc [hunk 2]
```diff
@@ -48,10 +47,10 @@ void KeyboardKeyHandler::KeyboardHook(int key,
   incoming->callback = std::move(callback);
 
   if (pending_responds_.size() > kMaxPendingEvents) {
-    std::cerr
+    FML_LOG(ERROR)
```
Comment (loic-sharma) [inline shell/platform/windows/keyboard_key_handler.cc:50]: Should this be a warning? Same comment for similar log in `shell/platform/windows/keyboard_manager.cc`

**Comment by loic-sharma** [comment]
@yaakovschectman It prepends the stream with a new line: https://github.com/flutter/engine/blob/643742e4fb97900e39b27c704e71bfbab8cce58b/fml/logging.cc#L75-L77

**Review by loic-sharma — APPROVED** [review]
LGTM with minor nits

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

File: shell/platform/windows/keyboard_key_handler.cc (hunk 2 above)
Comment (cbracken) [reply shell/platform/windows/keyboard_key_handler.cc:50]: Good question. If there are more responses than pending events, it suggests an invariant was violated. Error seems pretty reasonable to me.

**Pull request merged**